TestServeExecInContainerIdleTimeout is Flaky

I've had a few of my latest submits fail because of this test being flaky:

```
W0318 23:49:08.262989    2801 docker.go:728] found a container with the "k8s" prefix, but too few fields (2): "k8s_unidentified"
I0318 23:49:08.265061    2801 container_gc.go:104] Removing unidentified dead container "/k8s_unidentified" with ID "2876"
I0318 23:49:08.278875    2801 image_manager.go:240] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0318 23:49:08.280209    2801 image_manager.go:240] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0318 23:49:08.281372    2801 image_manager.go:240] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0318 23:49:08.282354    2801 image_manager.go:240] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0318 23:49:08.284508    2801 image_manager.go:186] [ImageManager]: Disk usage on "" () is at 95% which is over the high threshold (90%). Trying to free 150 bytes
I0318 23:49:08.291075    2801 image_manager.go:240] [ImageManager]: Removing image "image-0" to free 450 bytes
I0318 23:49:08.292083    2801 image_manager.go:186] [ImageManager]: Disk usage on "" () is at 95% which is over the high threshold (90%). Trying to free 150 bytes
I0318 23:49:08.292338    2801 image_manager.go:240] [ImageManager]: Removing image "image-0" to free 50 bytes
W0318 23:49:08.306996    2801 kubelet.go:321] Data dir for pod "bothpod" exists in both old and new form, using new
W0318 23:49:08.319499    2801 kubelet.go:361] Data dir for pod "newpod", container "bothctr" exists in both old and new form, using new
W0318 23:49:08.337750    2801 kubelet.go:1029] No ref for pod '1234'
W0318 23:49:08.342927    2801 kubelet.go:1029] No ref for pod '1234'
I0318 23:49:08.347996    2801 kubelet.go:1371] probe success: bar
I0318 23:49:08.353109    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
I0318 23:49:08.854629    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
I0318 23:49:08.868693    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
I0318 23:49:08.894820    2801 kubelet.go:1371] probe success: bar2
W0318 23:49:08.895567    2801 kubelet.go:1029] No ref for pod '1234'
I0318 23:49:08.895983    2801 kubelet.go:1512] Creating pod infra container for "foo1_new"
W0318 23:49:08.914872    2801 kubelet.go:1029] No ref for pod '1234'
W0318 23:49:08.915306    2801 kubelet.go:1029] No ref for pod '9876'
W0318 23:49:08.927940    2801 kubelet.go:1029] No ref for pod '1234'
W0318 23:49:08.928092    2801 kubelet.go:1029] No ref for pod '9876'
I0318 23:49:08.937546    2801 kubelet.go:2323] Query docker container info for pod bar_new failed with error (no containers exist for this pod)
I0318 23:49:08.938116    2801 kubelet.go:1371] probe success: foo
W0318 23:49:08.938226    2801 kubelet.go:1029] No ref for pod '1234'
I0318 23:49:08.938380    2801 kubelet.go:2323] Query docker container info for pod bar_new failed with error (no containers exist for this pod)
I0318 23:49:08.954379    2801 kubelet.go:2323] Query docker container info for pod foo_new failed with error (no containers exist for this pod)
I0318 23:49:08.955061    2801 kubelet.go:1380] pod "foo_new" container "bar" hash changed (4660 vs 358204163). Pod will be killed and re-created.
W0318 23:49:08.955343    2801 kubelet.go:1029] No ref for pod '9876'
W0318 23:49:08.955858    2801 kubelet.go:1029] No ref for pod '1234'
I0318 23:49:08.956096    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
I0318 23:49:08.964022    2801 kubelet.go:2323] Query docker container info for pod foo_new failed with error (no containers exist for this pod)
W0318 23:49:08.991855    2801 probe.go:147] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0318 23:49:08.992252    2801 probe.go:147] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0318 23:49:08.992532    2801 probe.go:147] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
I0318 23:49:08.992890    2801 kubelet.go:1376] pod "foo_new" container "bar" is unhealthy (probe result: unknown). Container will be killed and re-created.
W0318 23:49:08.993197    2801 kubelet.go:1029] No ref for pod '1234'
I0318 23:49:08.994545    2801 kubelet.go:2323] Query docker container info for pod foo_new failed with error (No pod infra container exists for this pod)
W0318 23:49:09.032094    2801 kubelet.go:570] Unknown protocol "": defaulting to TCP
W0318 23:49:09.033911    2801 kubelet.go:570] Unknown protocol "foobar": defaulting to TCP
E0318 23:49:09.116595    2801 kubelet.go:639] Invalid handler: &{<nil> <nil> <nil>}
I0318 23:49:09.132495    2801 kubelet.go:2323] Query docker container info for pod foo_new failed with error (no containers exist for this pod)
E0318 23:49:09.134217    2801 kubelet.go:1298] Error running pod "foo_new" container "bar": failed to call event handler: test error
I0318 23:49:09.134463    2801 kubelet.go:2323] Query docker container info for pod foo_new failed with error (no containers exist for this pod)
I0318 23:49:09.152971    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
E0318 23:49:09.174420    2801 kubelet.go:1442] Unable to generate status for pod with name "foo_new" and uid "12345678" info with error(Couldn't find spec for pod foo_new)
E0318 23:49:09.320293    2801 kubelet.go:1850] Pod "_": HostPort is already allocated, ignoring: [[0].port: duplicate value '81']
E0318 23:49:09.322227    2801 kubelet.go:1850] Pod "newpod_foo": HostPort is already allocated, ignoring: [[0].port: duplicate value '80']
E0318 23:49:09.329512    2801 kubelet.go:1850] Pod "_": HostPort is already allocated, ignoring: [[0].port: duplicate value '80']
E0318 23:49:09.413563    2801 kubelet.go:2094] error updating node status, will retry: error getting node testnode: Minions "testnode" not found
E0318 23:49:09.413739    2801 kubelet.go:2094] error updating node status, will retry: error getting node testnode: Minions "testnode" not found
E0318 23:49:09.413829    2801 kubelet.go:2094] error updating node status, will retry: error getting node testnode: Minions "testnode" not found
E0318 23:49:09.413938    2801 kubelet.go:2094] error updating node status, will retry: error getting node testnode: Minions "testnode" not found
E0318 23:49:09.414041    2801 kubelet.go:2094] error updating node status, will retry: error getting node testnode: Minions "testnode" not found
I0318 23:49:09.416919    2801 kubelet.go:2323] Query docker container info for pod bar_foo failed with error (no containers exist for this pod)
W0318 23:49:09.554258    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.554583    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.554700    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.555666    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.555771    2801 probe.go:147] Failed to find probe builder for  <nil>
W0318 23:49:09.555844    2801 probe.go:147] Failed to find probe builder for  <nil>
W0318 23:49:09.555908    2801 probe.go:147] Failed to find probe builder for  <nil>
W0318 23:49:09.555990    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.556180    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.556390    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.556551    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.556660    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.556747    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:100 TimeoutSeconds:0}
W0318 23:49:09.556861    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:100 TimeoutSeconds:0}
W0318 23:49:09.556975    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:100 TimeoutSeconds:0}
W0318 23:49:09.557108    2801 probe.go:59] No ref for pod 'foobar' - ''
W0318 23:49:09.557185    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.557284    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.557387    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.557537    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.557638    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0318 23:49:09.557743    2801 probe.go:147] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
I0318 23:49:09.568445    2801 runonce.go:62] waiting for 1 pods
I0318 23:49:09.568785    2801 runonce.go:139] container "bar" not found
I0318 23:49:09.568867    2801 runonce.go:112] pod "foo" containers not running: syncing
I0318 23:49:09.570685    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
E0318 23:49:09.584318    2801 kubelet.go:1442] Unable to generate status for pod with name "foo_new" and uid "12345678" info with error(Couldn't find spec for pod foo_new)
I0318 23:49:09.584510    2801 runonce.go:125] pod "foo" containers synced, waiting for 1ms
I0318 23:49:09.585970    2801 runonce.go:139] container "bar" not found
I0318 23:49:09.586045    2801 runonce.go:112] pod "foo" containers not running: syncing
W0318 23:49:09.587383    2801 docker.go:728] found a container with the "k8s" prefix, but too few fields (5): "k8s_net_foo.new.test_abcdefgh_42"
I0318 23:49:09.587719    2801 kubelet.go:1512] Creating pod infra container for "foo_new"
E0318 23:49:09.590339    2801 kubelet.go:1442] Unable to generate status for pod with name "foo_new" and uid "12345678" info with error(Couldn't find spec for pod foo_new)
I0318 23:49:09.590430    2801 runonce.go:125] pod "foo" containers synced, waiting for 2ms
I0318 23:49:09.593147    2801 runonce.go:108] pod "foo" containers running
I0318 23:49:09.593308    2801 runonce.go:75] started pod "foo"
I0318 23:49:09.593383    2801 runonce.go:84] 1 pods started
I0318 23:49:10.200987    2801 server.go:127] HTTP InternalServerError: Internal Error: pod level status currently unimplemented
E0318 23:49:12.291530    2801 server.go:618] Timed out waiting for client to create streams
--- FAIL: TestServeExecInContainerIdleTimeout (2.18 seconds)
	server_test.go:670: error creating input stream: Timeout occured
W0318 23:49:14.832974    2801 connection.go:126] Stream rejected: Unable to parse '' as a port: strconv.ParseUint: parsing "": invalid syntax
W0318 23:49:14.954697    2801 connection.go:126] Stream rejected: Unable to parse 'abc' as a port: strconv.ParseUint: parsing "abc": invalid syntax
W0318 23:49:15.115169    2801 connection.go:126] Stream rejected: Unable to parse '-1' as a port: strconv.ParseUint: parsing "-1": invalid syntax
W0318 23:49:15.215886    2801 connection.go:126] Stream rejected: Unable to parse '65536' as a port: strconv.ParseUint: parsing "65536": value out of range
W0318 23:49:15.307982    2801 connection.go:126] Stream rejected: Port '0' must be greater than 0
FAIL
coverage: 69.8% of statements
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet	9.254s
```

This test performs the following logic:

1. Test issues exec http get request to kubelet
1. Kubelet receives request
1. Kubelet sets idle timeout to 100ms
1. Kubelet waits up to 2 seconds for client to create necessary spdy streams (it will forcibly close the connection after 100ms of idleness)
1. Test tries to create a stream
1. Client waits up to 2 seconds for the server to acknowledge the stream
1. Test waits for connection to be closed by the server

Based on the log output above, the kubelet receives the client's exec request, but it doesn't receive a stream from the client. I'm guessing this is what's happening:

1. Server waits for a stream
1. > 100ms passes, so the connection is forcibly closed
1. Client attempts to create a stream
1. Server times out waiting for client to create a stream (`server.go:618] Timed out waiting for client to create streams`)
1. Client times out waiting for server to acknowledge stream creation (`server_test.go:670: error creating input stream: Timeout occurred`)

I would expect when the client attempts to create a stream, if the connection has been closed, that should be an immediate failure. It's possible that there's a race such that the client still thinks the connection is valid when it's really not.

I think a short-term fix is to just remove the test client's attempt to create a stream. The server will still close the connection after 100ms of idleness.

Another similar but not exactly the same stack trace

```
ok  	github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network/exec	2.033s	coverage: 96.7% of statements
W0409 19:01:21.730019   20173 docker.go:489] found a container with the "k8s" prefix, but too few fields (2): "k8s_unidentified"
I0409 19:01:21.730685   20173 container_gc.go:127] Removing unidentified dead container "/k8s_unidentified" with ID "2876"
I0409 19:01:21.752448   20173 image_manager.go:278] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0409 19:01:21.753314   20173 image_manager.go:278] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0409 19:01:21.760876   20173 image_manager.go:278] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0409 19:01:21.772583   20173 image_manager.go:278] [ImageManager]: Removing image "image-0" to free 1024 bytes
I0409 19:01:21.782626   20173 image_manager.go:216] [ImageManager]: Disk usage on "" () is at 95% which is over the high threshold (90%). Trying to free 150 bytes
I0409 19:01:21.792331   20173 image_manager.go:278] [ImageManager]: Removing image "image-0" to free 450 bytes
I0409 19:01:21.793191   20173 image_manager.go:216] [ImageManager]: Disk usage on "" () is at 95% which is over the high threshold (90%). Trying to free 150 bytes
I0409 19:01:21.793510   20173 image_manager.go:278] [ImageManager]: Removing image "image-0" to free 50 bytes
W0409 19:01:21.931729   20173 kubelet.go:436] Data dir for pod "bothpod" exists in both old and new form, using new
W0409 19:01:21.932948   20173 kubelet.go:491] Data dir for pod "newpod", container "bothctr" exists in both old and new form, using new
W0409 19:01:21.937083   20173 kubelet.go:976] No ref for pod '1234'
W0409 19:01:21.938106   20173 kubelet.go:976] No ref for pod '1234'
E0409 19:01:21.942941   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:21.947272   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:21.961626   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.017248   20173 kubelet.go:1916] error getting node: node testnode not found
I0409 19:01:22.019711   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:22.020576   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
E0409 19:01:22.164112   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.225043   20173 kubelet.go:1916] error getting node: node testnode not found
I0409 19:01:22.231978   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:22.232948   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
E0409 19:01:22.252369   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.310504   20173 kubelet.go:1916] error getting node: node testnode not found
I0409 19:01:22.456016   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:22.456797   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
E0409 19:01:22.469237   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.712192   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:22.740853   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.742273   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.745077   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:22.747550   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.750025   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.752116   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:22.761029   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.761697   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.763483   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
W0409 19:01:22.764434   20173 kubelet.go:976] No ref for pod '1234'
I0409 19:01:22.764857   20173 kubelet.go:1548] Creating pod infra container for "foo1_new"
E0409 19:01:22.826802   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:22.841429   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:22.847734   20173 kubelet.go:1916] error getting node: node testnode not found
W0409 19:01:22.848157   20173 kubelet.go:976] No ref for pod '9876'
W0409 19:01:22.848252   20173 kubelet.go:976] No ref for pod '1234'
E0409 19:01:22.849726   20173 kubelet.go:1916] error getting node: node testnode not found
W0409 19:01:22.849996   20173 kubelet.go:976] No ref for pod '1234'
W0409 19:01:22.850082   20173 kubelet.go:976] No ref for pod '9876'
I0409 19:01:23.222403   20173 kubelet.go:2404] Query docker container info for pod "bar_new" failed with error (no containers exist for this pod)
W0409 19:01:23.222911   20173 kubelet.go:976] No ref for pod '4567'
I0409 19:01:23.223015   20173 kubelet.go:2404] Query docker container info for pod "bar_new" failed with error (no containers exist for this pod)
I0409 19:01:23.227561   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:23.233029   20173 kubelet.go:1382] pod "foo_new" container "bar" hash changed (4660 vs 3317159511). Pod will be killed and re-created.
W0409 19:01:23.233918   20173 kubelet.go:976] No ref for pod '1234'
W0409 19:01:23.234080   20173 kubelet.go:976] No ref for pod '9876'
I0409 19:01:23.234367   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
E0409 19:01:23.237663   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
I0409 19:01:23.307322   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
W0409 19:01:23.308838   20173 probe.go:163] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0409 19:01:23.309042   20173 probe.go:163] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0409 19:01:23.309142   20173 probe.go:163] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
I0409 19:01:23.309258   20173 kubelet.go:1378] pod "foo_new" container "bar" is unhealthy (probe result: unknown). Container will be killed and re-created.
W0409 19:01:23.309349   20173 kubelet.go:976] No ref for pod '1234'
E0409 19:01:23.310110   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:23.802913   20173 kubelet.go:687] Invalid handler: &{<nil> <nil> <nil>}
I0409 19:01:23.807016   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
E0409 19:01:23.809446   20173 kubelet.go:1275] Error running pod "foo_new" container "bar": failed to call event handler: test error
I0409 19:01:23.811370   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
E0409 19:01:23.871325   20173 kubelet.go:1916] error getting node: node testnode not found
I0409 19:01:24.092408   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:24.094482   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
E0409 19:01:24.110043   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:24.762630   20173 kubelet.go:1882] Pod "_": HostPort is already allocated, ignoring: [[0].port: duplicate value '81']
E0409 19:01:24.817303   20173 kubelet.go:1882] Pod "newpod_foo": HostPort is already allocated, ignoring: [[0].port: duplicate value '80']
E0409 19:01:24.836975   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:24.909385   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:24.917997   20173 kubelet.go:1882] Pod "pod2_": HostPort is already allocated, ignoring: [[0].port: duplicate value '80']
E0409 19:01:24.932769   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:24.960308   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:25.353766   20173 kubelet.go:2150] error updating node status, will retry: error getting node "testnode": Node "testnode" not found
E0409 19:01:25.399792   20173 kubelet.go:2150] error updating node status, will retry: error getting node "testnode": Node "testnode" not found
E0409 19:01:25.400036   20173 kubelet.go:2150] error updating node status, will retry: error getting node "testnode": Node "testnode" not found
E0409 19:01:25.400160   20173 kubelet.go:2150] error updating node status, will retry: error getting node "testnode": Node "testnode" not found
E0409 19:01:25.400321   20173 kubelet.go:2150] error updating node status, will retry: error getting node "testnode": Node "testnode" not found
I0409 19:01:25.409658   20173 kubelet.go:2404] Query docker container info for pod "bar_foo" failed with error (no containers exist for this pod)
I0409 19:01:25.555772   20173 kubelet.go:2404] Query docker container info for pod "foo_ns" failed with error (no containers exist for this pod)
I0409 19:01:25.556568   20173 kubelet.go:1548] Creating pod infra container for "foo_ns"
E0409 19:01:25.580821   20173 kubelet.go:1590] Deleting mirror pod "foo_ns" because it is outdated
E0409 19:01:25.605339   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:25.628742   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:25.760359   20173 kubelet.go:1916] error getting node: node testnode not found
I0409 19:01:25.763029   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:25.780152   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
I0409 19:01:25.868952   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:25.879628   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
E0409 19:01:25.989237   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:25.996148   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:26.064571   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:26.089365   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:26.098236   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
I0409 19:01:26.140406   20173 kubelet.go:1210] Already successfully ran container "succeeded" of pod "foo_new", do nothing
E0409 19:01:26.158118   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
E0409 19:01:26.158891   20173 kubelet.go:1916] error getting node: node testnode not found
E0409 19:01:26.160755   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
I0409 19:01:26.161386   20173 kubelet.go:1201] Already ran container "succeeded" of pod "foo_new", do nothing
I0409 19:01:26.161988   20173 kubelet.go:1201] Already ran container "failed" of pod "foo_new", do nothing
W0409 19:01:26.162636   20173 kubelet.go:976] No ref for pod '9876'
E0409 19:01:26.163260   20173 kubelet.go:2438] Cannot get host IP: Cannot get node: node testnode not found
W0409 19:01:26.433648   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.434940   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.435006   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.435300   20173 probe.go:61] No ref for pod 'foobar' - ''
W0409 19:01:26.435357   20173 probe.go:163] Failed to find probe builder for  <nil>
W0409 19:01:26.435395   20173 probe.go:163] Failed to find probe builder for  <nil>
W0409 19:01:26.435438   20173 probe.go:163] Failed to find probe builder for  <nil>
W0409 19:01:26.435479   20173 probe.go:61] No ref for pod 'foobar' - ''
W0409 19:01:26.447979   20173 probe.go:61] No ref for pod 'foobar' - ''
W0409 19:01:26.448407   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:100 TimeoutSeconds:0}
W0409 19:01:26.448583   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:100 TimeoutSeconds:0}
W0409 19:01:26.448679   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:100 TimeoutSeconds:0}
W0409 19:01:26.448801   20173 probe.go:61] No ref for pod 'foobar' - ''
W0409 19:01:26.448867   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.448963   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.449045   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.449156   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.449241   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
W0409 19:01:26.449328   20173 probe.go:163] Failed to find probe builder for  &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:-100 TimeoutSeconds:0}
E0409 19:01:26.540995   20173 kubelet.go:1916] error getting node: node  not found
I0409 19:01:26.541663   20173 runonce.go:65] waiting for 1 pods
I0409 19:01:26.541951   20173 runonce.go:145] container "bar" not found
I0409 19:01:26.542028   20173 runonce.go:117] pod "foo" containers not running: syncing
I0409 19:01:26.608869   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:26.623936   20173 kubelet.go:1548] Creating pod infra container for "foo_new"
I0409 19:01:26.661904   20173 kubelet.go:2404] Query docker container info for pod "foo_new" failed with error (no containers exist for this pod)
I0409 19:01:26.662063   20173 runonce.go:130] pod "foo" containers synced, waiting for 1ms
W0409 19:01:26.694171   20173 docker.go:489] found a container with the "k8s" prefix, but too few fields (5): "k8s_net_foo.new.test_abcdefgh_42"
W0409 19:01:26.694511   20173 docker.go:603] Parse docker container name "/k8s_net_foo.new.test_abcdefgh_42" error: Docker container name "k8s_net_foo.new.test_abcdefgh_42" has less parts than expected [k8s net foo.new.test abcdefgh 42]
I0409 19:01:26.695854   20173 runonce.go:113] pod "foo" containers running
I0409 19:01:26.696088   20173 runonce.go:78] started pod "foo"
I0409 19:01:26.696234   20173 runonce.go:87] 1 pods started
I0409 19:01:27.547161   20173 server.go:155] HTTP InternalServerError: Internal Error: pod level status currently unimplemented
E0409 19:01:31.587462   20173 server.go:632] Timed out waiting for client to create streams
--- FAIL: TestServeExecInContainerIdleTimeout (3.63s)
	server_test.go:664: error creating input stream: use of closed network connection
```

--- a/pkg/kubelet/server.go
+++ b/pkg/kubelet/server.go
@@ -436,6 +436,8 @@ func (s *Server) handleRun(w http.ResponseWriter, req *http.Request) {
 	w.Write(data)
 }
 
+const streamCreationTimeout = 30 * time.Second
+
 // handleExec handles requests to run a command inside a container.
 func (s *Server) handleExec(w http.ResponseWriter, req *http.Request) {
 	u, err := url.ParseRequestURI(req.RequestURI)
@@ -491,9 +493,8 @@ func (s *Server) handleExec(w http.ResponseWriter, req *http.Request) {
 
 	conn.SetIdleTimeout(s.host.StreamingConnectionIdleTimeout())
 
-	// TODO find a good default timeout value
 	// TODO make it configurable?
-	expired := time.NewTimer(2 * time.Second)
+	expired := time.NewTimer(streamCreationTimeout)
 
 	var errorStream, stdinStream, stdoutStream, stderrStream httpstream.Stream
 	receivedStreams := 0
@@ -641,7 +642,7 @@ func waitForPortForwardDataStreamAndRun(pod string, uid types.UID, errorStream h
 
 	select {
 	case dataStream = <-dataStreamChan:
-	case <-time.After(1 * time.Second):
+	case <-time.After(streamCreationTimeout):
 		errorStream.Write([]byte("Timed out waiting for data stream"))
 		//TODO delete from dataStreamChans[port]
 		return

--- a/pkg/kubelet/server_test.go
+++ b/pkg/kubelet/server_test.go
@@ -655,15 +655,6 @@ func TestServeExecInContainerIdleTimeout(t *testing.T) {
 	if conn == nil {
 		t.Fatal("Unexpected nil connection")
 	}
-	defer conn.Close()
-
-	h := http.Header{}
-	h.Set(api.StreamType, api.StreamTypeError)
-	stream, err := conn.CreateStream(h)
-	if err != nil {
-		t.Fatalf("error creating input stream: %v", err)
-	}
-	defer stream.Reset()
 
 	<-conn.CloseChan()
 }

--- a/pkg/util/httpstream/spdy/connection.go
+++ b/pkg/util/httpstream/spdy/connection.go
@@ -71,7 +71,7 @@ func newConnection(conn *spdystream.Connection, newStreamHandler httpstream.NewS
 
 // createStreamResponseTimeout indicates how long to wait for the other side to
 // acknowledge the new stream before timing out.
-const createStreamResponseTimeout = 2 * time.Second
+const createStreamResponseTimeout = 30 * time.Second
 
 // Close first sends a reset for all of the connection's streams, and then
 // closes the underlying spdystream.Connection.

